agimus_controller/agimus_controller/warm_start_base.py

```diff
@@ -1,24 +1,69 @@
 from abc import ABC, abstractmethod
+
 import numpy as np
+import numpy.typing as npt
 
 from agimus_controller.trajectory import TrajectoryPoint
 
 
 class WarmStartBase(ABC):
+    """
+    A template class for implementing methods that generate a warmstart for an optimal control problem.
+    The warmstart is expected to generate the initial values for state and control trajectories,
+    based on the initial robot state and a reference trajectory.
+
+    Attributes:
+        _previous_solution (list[TrajectoryPoint]): Stores the previous solution of the optimization problem.
+
+    Methods:
+        generate(initial_state: TrajectoryPoint, reference_trajectory: list[TrajectoryPoint]) -> tuple:
+            Generates warm-start values for the optimization problem. This must be
+            implemented by subclasses.
+
+        update_previous_solution(previous_solution: list[TrajectoryPoint]) -> None:
+            Updates the internally stored previous solution for later use.
```

Comment:
Methods should have documentation coming from their own docstring. There is no need to document them at the global level